CMakeLists.txt

```diff
@@ -147,8 +147,8 @@ if (HDF5_USE_FOLDERS)
 endif ()
 option (HDF5_NO_PACKAGES "CPACK - Disable packaging" OFF)
 mark_as_advanced (HDF5_NO_PACKAGES)
-option (ALLOW_UNSUPPORTED "Allow unsupported combinations of configure options" OFF)
-mark_as_advanced (ALLOW_UNSUPPORTED)
+option (HDF5_ALLOW_UNSUPPORTED "Allow unsupported combinations of configure options" OFF)
```

Comment:
I'm not sure whether changing the CMake option names is something that should go into a minor release. It made sense for a 2.0 release, but for a minor release it's a bit disruptive.